### PR TITLE
improve docs and specs for `Cowboy.Handler`

### PR DESCRIPTION
**Pull request by beligante**
Hello Hello Folks!

I'm working on add a fix for: https://github.com/elixir-grpc/grpc/issues/184  and I had a hard time understanding the whole scope of all functions. While studying them, I thought it could be interesting add some specs, docs and love to them. This is mainly to help others to future contributions. Any feedback is welcomed.

### CODE REVIEW

File: lib/grpc/server/adapters/cowboy/handler.ex [hunk 1]
```diff
@@ -1,22 +1,53 @@
 defmodule GRPC.Server.Adapters.Cowboy.Handler do
   @moduledoc false
 
-  # A cowboy handler accepting all requests and calls corresponding functions
-  # defined by users.
+  # A cowboy handler accepting all requests and calls corresponding functions defined by users.
```
Comment (sleipnir) [inline lib/grpc/server/adapters/cowboy/handler.ex:4]:
```suggestion
  @moduledoc """
   A cowboy handler accepting all requests and calls corresponding functions defined by users.
   """
```

File: lib/grpc/server/adapters/cowboy/handler.ex [hunk 2]
```diff
@@ -1,22 +1,53 @@
 defmodule GRPC.Server.Adapters.Cowboy.Handler do
   @moduledoc false
```
Comment (sleipnir) [inline lib/grpc/server/adapters/cowboy/handler.ex:2]:
```suggestion
```

File: lib/grpc/server/adapters/cowboy/handler.ex [hunk 3]
```diff
@@ -116,54 +149,126 @@ defmodule GRPC.Server.Adapters.Cowboy.Handler do
...
+  @spec stream_body(
+          stream_pid :: pid,
+          data :: iodata,
+          opts :: list(stream_body_opts),
+          is_fin,
+          http_transcode? :: boolean()
+        ) :: :ok
+  def stream_body(pid, data, opts, is_fin, http_transcode? \\ false) do
+    send(pid, {:stream_body, data, opts, is_fin, http_transcode?})
+    :ok
```
[the remark this answers is not in the document]
Comment (sleipnir) [reply lib/grpc/server/adapters/cowboy/handler.ex:183]: Ok

**Review by sleipnir — APPROVED** [review]

File: lib/grpc/server/adapters/cowboy/handler.ex (hunk 1 above)
Comment (beligante) [reply lib/grpc/server/adapters/cowboy/handler.ex]: [c6cd103](https://github.com/elixir-grpc/grpc/pull/374/commits/c6cd103b798df319f5cae4e2af9bea9a28fa8fd5)

File: lib/grpc/server/adapters/cowboy/handler.ex (hunk 2 above)
Comment (beligante) [reply lib/grpc/server/adapters/cowboy/handler.ex]: [c6cd103](https://github.com/elixir-grpc/grpc/pull/374/commits/c6cd103b798df319f5cae4e2af9bea9a28fa8fd5)

**Comment by beligante** [comment]
> I approved with a small change request

Thanks a lot for the quick review

**Review by sleipnir — APPROVED** [review]

**Comment by sleipnir** [comment]
Thanks for working on this @beligante

**Pull request merged**